### PR TITLE
correct bug in negative binomial

### PR DESCRIPTION
**Pull request by annecori**
Fixing an error with 2 very problematic elements to it:

- one I that the over dispersion in the offspring distribution is wrong in the negative binomial version
- the other is that both the Negbin and Poisson projections will be wrong if instantaneous_R is true, because only one of the sampled R values was effectively used for forward simulation.
 
This was introduced in this [PR](https://github.com/reconhub/projections/pull/31/files) which by the way also introduces an incompatibility with EpiEstim in terms of how the Serial interval distribution is specified (starting at day 1 here and starting at day 0 in EpiEstim). 

The code is now not only wrong but really hard to read and I think we should discuss reverting all these changes for more clarity and ensuring everything is correct. Because it is so hard to read I haven't been able to double check all the changes introduced in this PR. 

These errors mean we will have to correct multiple papers that were already written and submitted to journals. @thibautjombart views on how to proceed with the code going forward would be welcome. 

Also, I don't understand how the tests are written for the negbin model so would appreciate your help @thibautjombart to change them so that this passes the tests. 

@pnouvellet has now double checked my suggested amendments and agrees with me. @stephaneghozzi tagging you FYI as the issue was introduced in a PR you made. 

Thanks

### CODE REVIEW

**Comment by stephaneghozzi** [comment]
I'm very sorry about that. I hope this will not have an impact on the conclusions of the work you did based on that, though you'll still have to re-run analyses.

(For reference here, Anne has flagged the point where I introduce the error in the computation of the size of the negative binomial: https://github.com/reconhub/projections/commit/6422906da17fb368a88b4c570955ce4770521e67#r101476925 
When redefining the variable `lambda` to include the multiplication by Rt, so that it corresponds to the mean of the Poisson distribution from which cases are drawn, I should have changed the formula in the line adjusting the size but instead just replaced the old `lambda` with the new one.)

**Comment by annecori** [comment]
Adding a comment for discussion with @thibautjombart  and @pnouvellet before we merge this - I think our fix only works for instantaneous_R = TRUE. Suggest at the moment to forbid using NegBin model when instantaneous_R = FALSE and work on how to fix the code for that case, unless you already have ideas of how to do that @stephaneghozzi ? thanks

**Comment by stephaneghozzi** [comment]
I see the option `instantaneous_R` was added later, so I haven't thought about that... and in any case I haven't looked at this code since then so it's surely better for people more familiar with the topic to look into it. 
Sorry if the code is hard to read, I could have a look and try and simplify and add some comments, but I'm not sure it would help lot.

**Comment by thibautjombart** [comment]
Following conversations with @annecori and @pnouvellet I would propose a fix adding a new function to calculate the size adjustment for the negbin separately. The function is called `compute_relative_infectivity`. After pulling the latest version of the branch of the PR, you can check the doc using: 
`?projections::compute_relative_infectivity`
(best displayed in rstudio)

I have not tested the fix, but keen to send asap for your input. If this is a sensible way to go, I will fix existing tests, add new tests for the new feature, and start prepping a CRAN release. Thanks all for flagging the issue, being so reactive, and finding a solution! :) 

**Please do not merge yet**

File: R/project.R [hunk 1]
```diff
@@ -346,12 +348,11 @@ project <- function(x, R, si, n_sim = 100, n_days = 7,
       ## mu will be 0 if lambda is 0. But that will make size 0 which
       ## will make rnbinom spit NAs. Workaround is: if lambda is 0
       ## set size to a non-trivial value.
-      size_adj <- lambda * size
+      size_adj <- compute_relative_infectivity(w = si, cases = out, t = i)
```
Comment (thibautjombart) [inline R/project.R:351]:
```suggestion
      size_adj <- compute_relative_infectivity(w = si, cases = out, t = i) * size
```

Comment (thibautjombart) [reply R/project.R:351]: Suggested by @annecori

File: R/compute_force_infection.R [hunk 2]
```diff
@@ -47,7 +47,7 @@ compute_force_infection <- function(w, cases, R, t, instantaneous_R) {
   if(!instantaneous_R) {
     lambda <- ws %*% (cases * R)
   } else {
-    lambda <- (ws %*% cases) * R[nrow(R)]
+    lambda <- (ws %*% cases) * R[nrow(R),]
```
Comment (annecori) [inline R/compute_force_infection.R:50]:
```suggestion
    lambda <- (ws %*% cases) * R[t,]
```

**Comment by pnouvellet** [comment]
Happy with changes! looks good, thanks for the corrections!

**Pull request merged**